agr_java_core/src/main/java/org/alliancegenome/core/translators/tdf/AlleleToTdfTranslator.java

```diff
@@ -30,24 +30,50 @@ public String getAllRows(List<Allele> annotations) {
 
     public List<AlleleDownloadRow> getAlleleDownloadRowsForGenes(List<Allele> annotations) {
 
+denormalizeAnnotations(annotations);
         return annotations.stream()
 
+                .filter(annotation -> annotation.getVariants()!=null)
+                .filter(annotation -> !CollectionUtils.isEmpty(annotation.getVariants()))
```

Comment:
Instead of 
!CollectionUtils.isEmpty(annotation.getVariants())
you can use the slightly easier-to-read version
CollectionUtils.isNotEmpty(annotation.getVariants())